examples/tensorflow-profiling-examples/trainer-deep-profiling/model.py

```diff
@@ -0,0 +1,151 @@
+        if mode == tf.estimator.ModeKeys.TRAIN:
+            dataset = dataset.batch(BATCH_SIZE, drop_remainder=True)
+        else:
+            dataset = dataset.batch(5012)
+        return dataset.repeat(num_epochs).make_one_shot_iterator().get_next()
+
+    return _input_fn
+
+
+def get_wide_deep(n_embeds):
+    # define column types
+    plurality_values = ['Single(1)', 'Twins(2)', 'Triplets(3)',
+                        'Quadruplets(4)', 'Quintuplets(5)', 'Multiple(2+)']
+    is_male = tf.feature_column.categorical_column_with_vocabulary_list('is_male', ['True', 'False', 'Unknown'])
+    mother_age = tf.feature_column.numeric_column('mother_age')
+    plurality = tf.feature_column.categorical_column_with_vocabulary_list('plurality', plurality_values)
+    gestation_weeks = tf.feature_column.numeric_column('gestation_weeks')
+
+    # discretize
+    age_buckets = tf.feature_column.bucketized_column(mother_age,
+                                                      boundaries=np.arange(15, 45, 1).tolist())  # pylint: disable=C0301
+    gestation_buckets = tf.feature_column.bucketized_column(gestation_weeks,
+                                                            boundaries=np.arange(17, 47, 1).tolist())  # pylint: disable=C0301
+
+    # sparse columns are wide
+    wide = [is_male, plurality, age_buckets, gestation_buckets]
+
+    # feature cross all the wide columns and embed into a lower dimension
+    crossed = tf.feature_column.crossed_column(wide, hash_bucket_size=20000)
+    embed = tf.feature_column.embedding_column(crossed, n_embeds)
+
+    # continuous columns are deep
+    deep = [mother_age, gestation_weeks, embed]
+    return wide, deep
+
+
+def my_rmse(labels, predictions):
+    pred_values = predictions['predictions']
+    return {'rmse': tf.metrics.root_mean_squared_error(labels, pred_values)}
+
+
+def serving_input_fn():
+    feature_placeholders = {'is_male': tf.placeholder(tf.string, [None]),
+                            'mother_age': tf.placeholder(tf.float32, [None]),
+                            'plurality': tf.placeholder(tf.string, [None]),
+                            'gestation_weeks': tf.placeholder(tf.float32, [None]),
+                            KEY_COLUMN: tf.placeholder_with_default(tf.constant(['nokey']), [None])  # pylint: disable=C0301
+                            }
+    features = {key: tf.expand_dims(tensor, -1)
+                for key, tensor in feature_placeholders.items()}
+    return tf.estimator.export.ServingInputReceiver(features,
+                                                    feature_placeholders)
+
+
+def train_and_evaluate(output_dir):
+    wide, deep = get_wide_deep(N_EMBEDS)
+    run_config = tf.estimator.RunConfig(save_checkpoints_secs=EVAL_INTERVAL,
+                                        keep_checkpoint_max=3)
+    hidden_units = [max(int(FIRST_LAYER_SIZE / (pow(2, i))), 2)
+                    for i in range(int(NUM_LAYERS))]
```

Comment:
NUM_LAYERS and FIRST_LAYER_SIZE are already int, no need for the int() cast

Reply:
second - ack
the first int() is still needed, since there are not guarantees that FIRST_LAYER_SIZE / (pow(2, i) would be integer for every hidden_layer